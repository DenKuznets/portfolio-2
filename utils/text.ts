const text = {
    showmore: 'Подробнее',
    backButton: 'Назад',
    myWorks: 'Мои проекты',
    allWorks: 'Все проекты',
    examples: 'Примеры работ',
    aboutMe: 'Обо мне',
    links: {
        inst: 'https://www.instagram.com/denis_kyznecov/',
        git: 'https://github.com/DenKuznets',
        tg: 'https://t.me/DenKuznets',
        project: '/projects/'
    },
    hero: {
        fname: 'Денис',
        lname: 'Кузнецов',
        profession: 'front-end разработчик',
        aboutme: 'верстаю сайты на react и styled-components',
        stack: 'СТЭК: react, nextjs, ts, mui, tailwind, styled-components, redux, rest, graphql, jest, react-testing-library, storybook',
        contactme: 'Связаться',
        socials: 'я в соц сетях'
    }
};

export default text;
