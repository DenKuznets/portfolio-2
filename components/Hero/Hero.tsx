import Image from 'next/image';
import text from '../../utils/text';
import instIcon from '../../public/images/instagram-logo.svg';
import gitIcon from '../../public/images/github-logo.svg';
import avatarPic from '../../public/images/avatar-violet-bg.webp';
import MyLink from '../MyLink/MyLink';
import Link from 'next/link';

export const hero_testids = {
    hero_container: 'hero_container'
};

const Hero = () => {
    const socials = [
        { link: text.links.git, icon: gitIcon },
        { link: text.links.inst, icon: instIcon }
    ];
    return (
        <div
            data-testid={hero_testids.hero_container}
            className="container md:mt-[90px] mx-auto flex flex-col items-center lg:grid lg:grid-cols-2 lg:justify-between lg:gap-8 uhd:max-w-[1920px]"
        >
            <div>
                <h1 className="font-bold text-5xl sm:text-6xl">
                    {text.hero.fname} {text.hero.lname}
                </h1>
                <div className="mt-1 flex items-center justify-end md:justify-start">
                    <hr className="mr-4 w-12 border-black md:w-20" />
                    <span className="first-letter:capitalize md:text-2xl">
                        {text.hero.profession}
                    </span>
                </div>
                <p className="color-black-900 mt-10 text-center md:text-start  uppercase leading-normal">
                    {text.hero.stack}
                </p>

                <div className="mt-14 flex flex-col items-center gap-4 md:flex-row ">
                    <MyLink violet href={text.links.tg}>
                        {text.hero.contactme}
                    </MyLink>
                    <MyLink className='whitespace-nowrap' href="/allworks">{text.myWorks}</MyLink>
                </div>
                <div className="flex justify-center items-center mt-14 md:justify-start">
                    <span className="first-letter:capitalize">
                        {text.hero.socials}:
                    </span>
                    <div className="flex px-4 gap-6">
                        {socials.map((obj) => (
                            <Link
                                className="hover:scale-125 transition-all"
                                key={obj.link}
                                href={obj.link}
                            >
                                <Image
                                    width={25}
                                    height={25}
                                    src={obj.icon}
                                    alt={obj.link}
                                />
                            </Link>
                        ))}
                    </div>
                </div>
            </div>
            <Image
                className="max-lg:hidden rounded-[30%_70%_70%_30%_/_30%_30%_70%_70%] shadow-[15px_15px_50px_rgba(0,0,0,0.2)]"
                width={700}
                height={900}
                src={avatarPic}
                alt="man in yellow jacket"
            />
        </div>
    );
};

export default Hero;
