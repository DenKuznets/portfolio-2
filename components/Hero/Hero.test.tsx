import { render, screen } from '@testing-library/react';
import Hero, { hero_testids } from './Hero';
import '@testing-library/jest-dom';
import text from '@/utils/text';

test('renders correctly', () => {
    render(<Hero />);
    const container = screen.getByTestId(hero_testids.hero_container);
    const h1 = screen.getByRole('heading', {level: 1})
    const p = screen.getByText(text.hero.stack)
    const telegramLink = screen.getByRole('link', { name: /связаться/i })
    const allProjects = screen.getByRole('link', {  name: /мои проекты/i})

    expect(container).toBeInTheDocument();
    expect(h1).toHaveTextContent(`${text.hero.fname} ${text.hero.lname}`)
    expect(p).toBeInTheDocument();
    expect(telegramLink).toHaveAttribute('href', 'https://t.me/DenKuznets')
    expect(allProjects).toHaveAttribute('href', '/allworks')
});

// test("renders hero unchanged", () => {
//   const { container } = render(<Hero />);
//   expect(container).toMatchSnapshot();
// });
