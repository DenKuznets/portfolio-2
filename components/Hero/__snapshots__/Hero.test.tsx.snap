// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders hero unchanged 1`] = `
<div>
  <div
    class="container mx-auto flex flex-col items-center p-2 md:p-4 lg:grid lg:grid-cols-2 lg:justify-between lg:gap-8"
    data-testid="hero_container"
  >
    <div
      data-testid="hero_text"
    >
      <h1
        class="font-bold"
      >
        Денис
         
        Кузнецов
      </h1>
      <div
        class="mt-1 flex items-center justify-end md:justify-start"
      >
        <hr
          class="mr-4 w-12 border-black md:w-20"
        />
        <span
          class="first-letter:capitalize md:text-2xl"
        >
          front-end разработчик
        </span>
      </div>
      <p
        class="color-black-900 mt-10 text-center  uppercase leading-normal"
      >
        СТЭК: react, ts, mui, styled-components, redux, rest, graphql (apollo-client), jest, react-testing-library, storybook
      </p>
      <div
        class="mt-14 flex flex-col items-center gap-4 md:flex-row "
        data-testid="hero_text_links"
      >
        <a
          class=" bg-violet text-white hover:text-amber-400 [&>svg]:hover:fill-amber-400 hover:outline-0 outline-0 border-none border-none outline-none transition-all hover:outline hover:outline-1 flex w-fit items-center rounded-2xl hover:outline-black  px-10 py-5 text-black  [&>svg]:hover:rotate-45 [&>svg]:ml-2 [&>svg]:transition-all"
          href="https://t.me/DenKuznets"
        >
          Связаться
          <svg
            fill="white"
            height="30px"
            viewBox="0 0 24 24"
            width="30px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              data-name="Location Arrow 1"
              id="Location_Arrow_1"
            >
              <path
                d="M14.472,20.937a1.438,1.438,0,0,1-1.3-.812L10.3,14.343a1.418,1.418,0,0,0-.642-.641L3.874,10.831A1.462,1.462,0,0,1,4.06,8.136l14.952-5a1.46,1.46,0,0,1,1.849,1.847l-5,14.952a1.439,1.439,0,0,1-1.284.994C14.543,20.936,14.507,20.937,14.472,20.937ZM19.479,4.063a.488.488,0,0,0-.149.024h0l-14.952,5a.46.46,0,0,0-.058.849L10.1,12.805A2.444,2.444,0,0,1,11.2,13.9l2.87,5.782a.443.443,0,0,0,.445.255.45.45,0,0,0,.4-.312l5-14.953a.462.462,0,0,0-.433-.607Z"
              />
            </g>
          </svg>
        </a>
        <a
          class="  border-none outline-none transition-all hover:outline hover:outline-1 flex w-fit items-center rounded-2xl hover:outline-black  px-10 py-5 text-black  [&>svg]:hover:rotate-45 [&>svg]:ml-2 [&>svg]:transition-all"
          href="https://t.me/DenKuznets"
        >
          Мои проекты
          <svg
            fill="none"
            height="30px"
            viewBox="0 0 24 24"
            width="30px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 17L17 7M17 7H8M17 7V16"
              stroke="#000000"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </a>
      </div>
      <div
        class="flex justify-center items-center mt-14 md:justify-start"
        data-testid="hero_text_socials"
      >
        <span
          class="first-letter:capitalize"
        >
          я в соц сетях
          :
        </span>
        <div
          class="flex"
        >
          <a
            href="https://www.instagram.com/denis_kyznecov/"
          >
            <div
              class="social-icon-container"
            >
              <img
                alt="instagram-logo"
                data-nimg="1"
                decoding="async"
                height="25"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
                style="color: transparent;"
                width="25"
              />
            </div>
          </a>
          <a
            href="https://github.com/DenKuznets"
          >
            <div
              class="social-icon-container"
            >
              <img
                alt="github-logo"
                data-nimg="1"
                decoding="async"
                height="25"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
                style="color: transparent;"
                width="25"
              />
            </div>
          </a>
        </div>
      </div>
    </div>
    <img
      alt="man in yellow jacket"
      class="max-lg:hidden rounded-[30%_70%_70%_30%_/_30%_30%_70%_70%] shadow-[15px_15px_50px_rgba(0,0,0,0.2)]"
      data-nimg="1"
      data-testid="hero_image"
      decoding="async"
      height="900"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=1920&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=750&q=75 1x, /_next/image?url=%2Fimg.jpg&w=1920&q=75 2x"
      style="color: transparent;"
      width="700"
    />
  </div>
</div>
`;
